crate/operator/webhooks.py

```diff
@@ -37,6 +37,7 @@ class WebhookEvent(str, enum.Enum):
     HEALTH = "health"
     FEEDBACK = "feedback"
     BACKUP_SCHEDULE_UPDATE = "backup_schedule_update"
+    SNAPSHOT_RESTORED = "snapshot_restored"
```

Comment:
This also needs handling in the API, right? How did it work before?

Reply:
At the moment it works with the `FEEDBACK` event type only. This type only supports `operation` and `message`, so I think it's a good idea to introduce a new event type for reporting back the admin username. But I think we should not mix notifications for the status of the restore operation and the admin username. With [this notification](https://github.com/crate/crate-operator/pull/451/files#diff-71c803b5c41e32172f7ec518522689fc1da8582141d29c6c26c61c32af88f07eR604) and the API changes in the other PR, we would mark the whole restore operation as succeeded, right? Although `RestoreSystemUserPasswordSubHandler` runs even if the restore itself failed (`run_on_dep_failures=True`)

What if we keep the status updates (also in the API) as they are with type `FEEDBACK` and just move this new part with the admin user check and the new notification to `SendSuccessNotificationSubHandler`? Then it would only update the CRD and send the webhook if everything went well. So I mean `SendSuccessNotificationSubHandler` would then schedule 2 notifications, the existing status success and a new one with type `SNAPSHOT_RESTORED` and the API would only update the cluster's username for the new event.

Reply:
It used to work with the feedback webhook, which just sends a string back

Reply:
Thanks for the clarification @tomach , I think that's a great plan. Then perhaps ``SNAPSHOT_RESTORED`` is not a very descriptive name, how about we call it something more descriptive like ``ADMIN_USERNAME_CHANGED``?

Reply:
:+1: sounds good